src/app/plugins/dog.plugin.ts

```diff
@@ -1,66 +1,133 @@
-      breed = this._breeds[this._getRandomInt(this._breeds.length)];
+    if (breed.startsWith('subbreed')) {
+      await message.reply(await this._makeSubBreedEmbed());
+      return;
     }
 
-    if (!breed) {
-      this.container.loggerService.warn(`Dog API down.`);
-      await message.reply(`Sorry, the dog plugin has encountered an error.`);
+    if (breed.startsWith('breed')) {
+      await message.reply(await this._makeBreedEmbed());
       return;
     }
 
-    if (!this._breeds.includes(breed)) {
-      message.reply(`Breeds supported: \n\`\`\`\n${this._breeds.join('\n')}\`\`\``);
-      return;
+    let url = `breed/${breed.replace(' ', '/')}/images/random`; // The / slash is just how the API takes in the sub breeds
+
+    if (breed === '' || breed === 'random') {
+      url = 'breeds/image/random';
     }
 
     await this.container.httpService
-      .get(`${this._API_URL}breeds/${breed}/image`)
+      .get(`${this._API_URL}${url}`)
       .then((response: IHttpResponse) => {
-        if (!!response.data.response.url) {
-          message.reply('', {
-            files: [response.data.response.url],
-          });
+        // notifies the user if their was a problem contacting the server
+        if (Math.floor(response.status / 100) != 2) {
+          message.reply(
+            `Something seems to have happened with the connection to ${this._API_URL}.`
+          );
+          return;
         }
+
+        message.reply('', {
+          files: [response.data.message],
+        });
       })
-      .catch((err) => this.container.loggerService.warn(err));
+      .catch((err) => {
+        this.container.loggerService.warn(err);
+      });
   }
 
-  private _getRandomInt(max: number): number {
-    return Math.floor(Math.random() * Math.floor(max));
+  async _makeBreedEmbed(): Promise<MessageEmbed> {
+    if (this._breedEmbed) {
+      return this._breedEmbed;
+    }
+
+    this._breedEmbed = new MessageEmbed();
+
+    this._breedEmbed = await this.container.messageService.generateEmbedList(this._breeds, {
+      title: 'Sub Breeds',
+    });
+
+    return this._breedEmbed;
+  }
+
+  async _makeSubBreedEmbed(): Promise<MessageEmbed> {
+    if (this._subBreedEmbed) {
+      return this._subBreedEmbed;
+    }
+
+    const embed = new MessageEmbed();
+    embed.setColor('#0099ff').setTitle('Sub Breeds');
+
+    this._subBreeds.forEach((breed) => {
+      embed.addField(breed.breed, breed.subBreed.join('\n'), true);
+    });
+
+    return (this._subBreedEmbed = embed);
   }
 
-  private _parseBreed(args: string[]): string {
+  private _parseInput(args: string[]): string {
     return args.map((str) => str.toLowerCase()).join(' ');
   }
+
+  public validate(message: IMessage, args?: string[]) {
+    const input = this._parseInput(args || []);
+
+    const hasKeyword: boolean = this._VALID_KEYS.includes(input);
+    const hasSub: boolean = this._allBreeds.includes(input);
+
+    return hasSub || hasKeyword;
+  }
```

Comment:
If the user inputs an incorrect breed, it wont be obvious that's what happened

Reply:
Should probably check if the breed exists from execute so the user knows they atleast put correct args

Reply:
it does in parse

Reply:
sorry validate*

Reply:
Yeah, you should do that *after* validate since validate is called before execute
bc if the user puts an incorrect breed, it will say *@Tanndlin, Invalid arguments! Try: !dog <breed (Optional)> | dog breed | dog subBreed | dog random*